Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -3720,6 +3728,12 @@ LayoutUnit RenderBox::constrainBlockMarginInAvailableSpaceOrTrim(const RenderBox
     return marginSide == MarginTrimType::BlockStart ? minimumValueForLength(style().marginBeforeUsing(&containingBlock.style()), availableSpace) : minimumValueForLength(style().marginAfterUsing(&containingBlock.style()), availableSpace);
 }
 
+bool RenderBox::hasTrimmedBlockStartMarginInBlockContainer() const
```

Comment:
Does "inBlockContainer" add any value? I would assume the caller only cares whether there's trimmed margin or not.

Reply:
Currently it's to make clear that it is only checking in the context of a block container and I was going to modify the name once the flexbox and grid portions were implemented. Should I instead change the name now and add some checks for `isFlexItem()` and `isGridItem()` along with some comments inside the function?

Reply:
You may find it hard to come up with a name when e.g. flex is added but not yet grid. In cases like this, I'd make the decision at the callsite and assert in this function if it happens to be called on unsupported content.

Reply:
Yeah that's reasonable as well. I did a few things to refactor/clean this up:
- Changed the name to `hasTrimmedMargin` and takes in an argument that specifies the margin to check
- Add logic to check supported content (layout context and type of margin trim specified) and assert when it gets called with something unsupported
- Modified the call inside `ComputedStyleExtractor` to only call this function on items that are not flex or grid items